src/presetsController.ts

```diff
@@ -42,12 +43,26 @@ export class PresetsController {
         const presetsController = new PresetsController(project, kitsController, isMultiProject);
         const expandSourceDir = async (dir: string) => {
             const workspaceFolder = project.workspaceFolder.uri.fsPath;
+            const kit = project.getActiveKit();
```

Comment:
This part worries me, if we're in the presetsController then don't we not care about kits? Since it's either kits or Presets? Not both?

Reply:
I didn't step back and realize this obvious fact. I just noticed errors and complains in my repro and just went head first into making those go away. I'll rethink a bit and get back with an update. Very good essential point.